source/fundamentals/crud/write-operations/bulk-write.txt

```diff
@@ -0,0 +1,546 @@
+The operation matches and deletes the first document
+where the value of the ``restaurant_id`` field is ``"5678"``.
+
+.. literalinclude:: /includes/fundamentals/code-examples/BulkWrite.cs
+   :start-after: start-bulk-delete-one
+   :end-before: end-bulk-delete-one
+   :language: csharp
+   :copyable:
+   :dedent: 8
+
+To delete multiple documents, create an instance of the ``BulkWriteDeleteManyModel<TDocument>``
+class and pass a query filter that specifies the document that you want to delete.
+The ``DeleteMany`` operation removes *all documents* that match your
+query filter.
+
+In the following code example, the ``BulkWriteDeleteManyModel<BsonDocument>`` object
+represents a delete operation on the ``sample_restaurants.restaurants`` collection.
+The operation matches and deletes all documents
+where the value of the ``name`` field is ``"Mongo's Deli"``.
+
+.. literalinclude:: /includes/fundamentals/code-examples/BulkWrite.cs
+   :start-after: start-bulk-delete-many
+   :end-before: end-bulk-delete-many
+   :language: csharp
+   :copyable:
+   :dedent: 8
+
+Run the Write Operations
+------------------------
+
+After you define a ``BulkWriteModel`` instance for each operation that you want to perform,
+create an instance of a class that implements the ``IReadOnlyList`` interface. Add your
+``BulkWriteModel`` objects to this ``IReadOnlyList``, then pass the ``IReadOnlyList``
+to the ``BulkWrite()`` or ``BulkWriteAsync()`` method. By default, these methods run
+the operations in the order they're defined in the collection.
+
+.. tip:: IReadOnlyList
+
+   ``Array`` and ``List`` are two common classes that implement the ``IReadOnlyList``
+   interface.
+
+Select from the following tabs to view how to use the asynchronous
+``BulkWriteAsync()`` method and the synchronous ``BulkWrite()`` method to
+perform a bulk write operation.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: bulk-write-async
+
+      .. literalinclude:: /includes/fundamentals/code-examples/BulkWrite.cs
+         :start-after: start-bulk-write-async
+         :end-before: end-bulk-write-async
+         :language: csharp
+         :copyable:
+         :dedent: 8
+
+   .. tab:: Synchronous
+      :tabid: bulk-write-sync
+
+      .. literalinclude:: /includes/fundamentals/code-examples/BulkWrite.cs
+         :start-after: start-bulk-write-sync
+         :end-before: end-bulk-write-sync
+         :language: csharp
+         :copyable:
+         :dedent: 8
+
+The preceding code examples produce the following output:
+
+.. code-block:: shell
+      
+   BulkWriteResult({'writeErrors': [], 'writeConcernErrors': [], 'nInserted': 2, 'nUpserted': 0, 'nMatched': 2, 'nModified': 2, 'nRemoved': 1, 'upserted': []}, acknowledged=True)
+
+.. tip::
+    
+    If any of the write operations fail, the {+driver-short+} raises a
+    ``BulkWriteException`` and does not perform any further operations. You can examine
```

Comment:
It should be [ClientBulkWriteException](https://mongodb.github.io/mongo-csharp-driver/3.0.0/api/MongoDB.Driver/MongoDB.Driver.ClientBulkWriteException.html) instead.